AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -0,0 +1,87 @@
+from consts.scripting import (VARIABLE_MANAGER_QT, VARIABLE_TYPES, VARIABLE_PALETTE_QT, ADD_BUTTON_QT, WAIT_TIME_3,
+                              GRAPH_VARIABLES_QT)
+from consts.general import (WAIT_TIME_3, VARIABLE_TYPES_DICT)
+from editor_python_test_tools.utils import Report
+
+
+class QtPyScriptCanvasVariableManager:
+    """
+    QtPy class for handling the behavior of the script canvas variable manager
+    """
+
+    def __init__(self, sc_editor):
+        self.variable_manager = sc_editor.findChild(QtWidgets.QDockWidget, VARIABLE_MANAGER_QT)
+
+    def create_new_variable(self, new_variable_type):
+        """
+        function for adding a new variable to the variable manager's list
+
+        param new_variable_type: the type of variable you want to create provided as a string. Is case-sensitive!
+
+        returns: None
+        """
+        self.__validate_new_variable(new_variable_type)
+
+        add_new_variable_button = self.variable_manager.findChild(QtWidgets.QPushButton, ADD_BUTTON_QT)
+        add_new_variable_button.click()  # Click on Create Variable button
+
+        helper.wait_for_condition((
+            lambda: self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT) is not None), WAIT_TIME_3)
+
+        # Select variable type
+        table_view = self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT)
+        model_index = pyside_utils.find_child_by_pattern(table_view, new_variable_type)
+
+        # Click on it to create variable
+        pyside_utils.item_view_index_mouse_click(table_view, model_index)
+
+    def __validate_new_variable(self, new_variable_type):
+        """
+        function for checking the provided variable type for validity
+        """
+        if type(new_variable_type) is not str:
+            Report.critical_result(["Invalid variable type provided", ""], False)
+
+        valid_type = False
+        for this_type in VARIABLE_TYPES:
+            if new_variable_type == this_type:
+                valid_type = True
+
+        if not valid_type:
+            Report.critical_result(["Invalid variable type provided", ""], False)
+
+    def get_basic_variable_types(self):
+        """
+        function for getting easy to use container of variable types off the general constants file
+
+        returns: simple namespace container that allows you to access the most common variable types (strings) using dot notation
+        """
+        simple_namespace = SimpleNamespace(**VARIABLE_TYPES_DICT)
+
+        return simple_namespace
+
+    def get_variable_count(self):
+        """
+        function for retrieving the number of variables in the variable manager's list
+
+        returns: the number of variables as an integer
+        """
+        graph_variables = self.variable_manager.findChild(QtWidgets.QTableView, GRAPH_VARIABLES_QT)
+        row_count = graph_variables.model().rowCount(QtCore.QModelIndex())
+        return row_count
```

Comment:
Whitespace.

You can also `return graph_variables.model().rowCount(QtCore.QModelIndex())`.

Reply:
Feedback still open.

Reply:
I think I'll keep it this way for now. I prefer to return a variable if possible instead of inline function calls.